pkg/minikube/constants/constants.go

```diff
@@ -45,9 +46,17 @@ const MinikubeContext = "minikube"
 // MinikubeEnvPrefix is the prefix for the environmental variables
 const MinikubeEnvPrefix = "MINIKUBE"
 
+// GetMinipath gets the path for the .minikube directory
+func GetMinipath() string {
+	if filepath.Base(viper.GetString("Home")) == ".minikube" { // "Home" value comes from config.MinikubeHome
+		return viper.GetString("Home")
+	}
+	return filepath.Join(viper.GetString("Home"), ".minikube")
+}
```

Comment:
pkg/minikube should not reference spf13/viper, and since this isn't a constant anymore - i don't think theres any reason for it to live here.

How about the information comes straight from config.MinikubeHome?  The only time where it looks like you don't have this information is in the Downloader (my oversight).  You could add it as a field member when it gets added into the config.